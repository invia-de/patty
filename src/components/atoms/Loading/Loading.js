import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../utils/classnames';
import styles from './loading.module.scss';

export default function Loading({ children, className, size }) {
  return (
    <div className={cx(styles.loading, className, styles[size])}>
      <svg className={styles.svg} x="0" y="0" viewBox="0 0 150 150">
        <circle className={styles.circle} cx="75" cy="75" r="60" />
      </svg>
      {children}
    </div>
  );
}
Loading.propTypes = {
  /** size */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /** additional classname you want to add */
  className: PropTypes.string,
  /** use children to display text next to the spinner */
  children: PropTypes.node
};
Loading.defaultProps = {
  size: 'medium'
};
