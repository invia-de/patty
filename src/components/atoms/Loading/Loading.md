```jsx
<Loading />
<Loading size="large" />
<Loading size="medium" />
<Loading size="small" />
```
