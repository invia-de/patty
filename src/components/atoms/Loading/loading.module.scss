@import '../../../style/colors';

.loading {
  text-align: center;
}

.svg {
  width: 1em;
  height: 1em;
  vertical-align: bottom;
}

.small {
  font-size: 1rem;
}

.medium {
  font-size: 1.5rem;
}

.large {
  font-size: 2.5rem;
}

.circle {
  stroke-dashoffset: 0;
  stroke-dasharray: 300;
  stroke-width: 1.25rem;
  stroke-miterlimit: 15;
  stroke-linecap: square;
  transform-origin: center center;
  animation: loading 2s linear infinite;
  stroke: $aqua;
  fill: transparent;
}

@keyframes loading {
  0% {
    transform: rotate(0);
    stroke-dashoffset: 0;
  }
  100% {
    transform: rotate(360deg);
    stroke-dashoffset: -600;
  }
}
