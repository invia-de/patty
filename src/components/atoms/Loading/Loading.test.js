import React from 'react';
import Loading from './Loading';
import { render } from '@testing-library/react';

test('Loading renders', () => {
  const { container } = render(<Loading />);
  expect(container.firstChild).toMatchSnapshot();
});
