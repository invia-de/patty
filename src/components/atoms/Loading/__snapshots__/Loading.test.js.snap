// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders 1`] = `
<div
  class="loading medium"
>
  <svg
    class="svg"
    viewBox="0 0 150 150"
    x="0"
    y="0"
  >
    <circle
      class="circle"
      cx="75"
      cy="75"
      r="60"
    />
  </svg>
</div>
`;
